src/components/rule-builder/rule-builder.component.tsx

```diff
@@ -0,0 +1,1028 @@
+              items={answer}
+              onChange={handleValueChange}
+              size={isTablet ? 'lg' : 'sm'}
+            />
+          )}
+        </div>
+        <Layer className={styles.layer}>
+          <OverflowMenu
+            aria-label={t('optionsMenu', 'Options Menu')}
+            className={styles.overflowMenu}
+            size={isTablet ? 'lg' : 'sm'}
+            align="left"
+            flipped
+          >
+            <OverflowMenuItem
+              className={styles.menuItem}
+              id="addCondition"
+              onClick={addCondition}
+              itemText={t('addCondition', 'Add condition')}
+              hasDivider
+            />
+            {!isNewCondition ? (
+              <OverflowMenuItem
+                className={styles.menuItem}
+                id="addNewConditionalLogic"
+                onClick={addNewConditionalLogic}
+                itemText={t('addNewLogic', 'Add new logic')}
+              />
+            ) : (
+              <OverflowMenuItem
+                className={styles.menuItem}
+                id="deleteCondition"
+                onClick={deleteCondition}
+                itemText={t('deleteCondition', 'Delete condition')}
+                hasDivider
+                isDelete
+              />
+            )}
+            {!isNewRule && !isNewCondition && (
+              <OverflowMenuItem
+                className={styles.menuItem}
+                id="deleteConditionalLogic"
+                onClick={() => launchDeleteConditionalLogicModal(true)}
+                itemText={t('deleteAllLogics', 'Delete conditional logic')}
+                hasDivider
+                isDelete
+              />
+            )}
+            {!isNewCondition && (
+              <OverflowMenuItem
+                className={styles.menuItem}
+                id="deleteSingleConditionalLogic"
+                onClick={() => launchDeleteConditionalLogicModal(false)}
+                itemText={t('deleteSingleLogic', 'Delete')}
+                hasDivider
+                isDelete
+              />
+            )}
+          </OverflowMenu>
+        </Layer>
+      </div>
+    );
+  },
+);
+
+interface RuleActionProps {
+  fieldId: string;
+  questions: Array<Question>;
+  isTablet: boolean;
+  deleteAction: () => void;
+  isNewAction: boolean;
+  addAction: () => void;
+  index: number;
+  handleActionChange: (id: string, field: string, value: string, index: number) => void;
+  actions: Array<Action>;
+}
+
+export const RuleAction = React.memo(
+  ({
+    fieldId,
+    isTablet,
+    questions,
+    index,
+    isNewAction,
+    deleteAction,
+    addAction,
+    handleActionChange,
+    actions,
+  }: RuleActionProps) => {
+    const { t } = useTranslation();
+    const [action, setAction] = useState<string>('');
+    const [errorMessage, setErrorMessage] = useState<string>(actions[index]?.errorMessage || '');
+    const [isCalculate, setIsCalculate] = useState<boolean>(false);
```

Comment:
Boolean and string annotations can be inferred from usage so we can safely omit them here.